lib/storage/providers/ProviderInterface.js

```diff
@@ -0,0 +1,75 @@
+function notImplemented() {
+    throw new Error('Method not implemented');
+}
+
+/**
+ * @interface
+ * The interface Onyx uses to communicate with the storage layer
+ */
+class ProviderInterface {
```

Comment:
This can serve as an abstract class and even provide some base functionality like parsing/stringifying data 

ATM no-one is extending or using the class created here it serves for documentation purposes 
- declaring a new Provider and annotating it with `@implements {ProviderInterface}` would prompt to implement all methods (Webstorm ®)
- Providers will also inherit all method documentation (and param types)

Reply:
>no-one is extending or using the class created here it serves for documentation purposes

Can we maybe explore this when there is a clear need? It feels like an unnecessary distraction for the current exercise.